shared-module/keypad/EventQueue.c

```diff
@@ -96,3 +96,8 @@ bool keypad_eventqueue_record(keypad_eventqueue_obj_t *self, mp_uint_t key_numbe
 
     return true;
 }
+
+bool common_hal_keypad_eventqueue_get_available(keypad_eventqueue_obj_t *self) {
+    int encoded_event = ringbuf_peek16(&self->encoded_events);
+    return encoded_event != -1;
```

Comment:
`EventQueue` already implements `__bool__()`, which returns False if the queue length is zero. It calls `common_hal_keypad_eventqueue_get_length()` to figure that out. So could you use that existing functionality instead in your sample async class/